tests/context_tests.rs

```diff
@@ -366,28 +366,26 @@ mod test_pcr_extend_reset {
         // get reset by any D-RTPM events.
         // PCR (TCG PC Client Platform TPM Profile (PTP) for TPM 2.0 Version 1.05 Rev 14)
         let mut context = create_ctx_with_session();
-        let pcr_ses = context.sessions();
+        let pcr_ses = context.sessions().0;
 
         // We start by resetting. We do not place any expectations on the prior contents
-        context.pcr_reset(PcrHandle::Pcr16).unwrap();
+        context.execute_with_session(pcr_ses, |ctx| ctx.pcr_reset(PcrHandle::Pcr16).unwrap());
```

Comment:
Looks like you're only using this `pcr_ses` session now, wouldn't it be easier to simply set it as the main session on `Context` and use the new methods only where you need a different one?

Reply:
I personally prefer to use the `execute_with_session` everywhere, to make it explicit which sessions are used with which calls.

Reply:
If you prefer to use the `pcr_ses` by default, and only `execute_with_session` in the special case, I'll happily change. This is just my personal preference of explicit over implicit.

Reply:
I don't mind either way, it's only a test! It does help with checking that the thing works properly 😄 So leave as is!